diff --git a/BUILD b/BUILD
index 2fb2605..7861c43 100644
--- a/BUILD
+++ b/BUILD
@@ -4,6 +4,7 @@ licenses(["notice"])
 
 exports_files(["LICENSE"])
 
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
 ################################################################################
 # Java 9 configuration
 ################################################################################
@@ -19,7 +20,7 @@ config_setting(
 # ZLIB configuration
 ################################################################################
 
-ZLIB_DEPS = ["@zlib//:zlib"]
+ZLIB_DEPS = ["@zlib_archive//:zlib"]
 
 ################################################################################
 # Protobuf Runtime Library
@@ -218,7 +219,7 @@ cc_library(
 # TODO(keveman): Remove this target once the support gets added to Bazel.
 cc_library(
     name = "protobuf_headers",
-    hdrs = glob(["src/**/*.h"]),
+    hdrs = glob(["src/**/*.h", "src/**/*.inc"]),
     includes = ["src/"],
     visibility = ["//visibility:public"],
 )
@@ -674,8 +675,6 @@ py_library(
             "python/google/protobuf/**/*.py",
         ],
         exclude = [
-            "python/google/protobuf/__init__.py",
-            "python/google/protobuf/**/__init__.py",
             "python/google/protobuf/internal/*_test.py",
             "python/google/protobuf/internal/test_util.py",
         ],
@@ -836,6 +835,19 @@ py_library(
     ],
 )
 
+pkg_tar(
+    name = "protobuf_python_package",
+    package_dir = "usr/local/lib/python/site-packages/",
+    extension = "tar",
+    srcs = [
+        ":python_srcs",
+        ":protobuf_python",
+    ],
+    strip_prefix = "python",
+    tags = ["manual"],
+    visibility = ["//visibility:public"],
+)
+
 internal_protobuf_py_tests(
     name = "python_tests_batch",
     data = glob([
